src/controllers/user.controller.js

```diff
@@ -344,19 +371,42 @@ export const updateProfile = async (req, res) => {
       }
     }
 
-
-    Object.keys(updates).forEach((key) => {
-      if (updates[key] !== undefined) {
-        user[key] = updates[key];
-      }
-    });
+    // Update user fields with the new values
+    user.firstName = updates.firstName !== undefined ? updates.firstName : user.firstName;
+    user.lastName = updates.lastName !== undefined ? updates.lastName : user.lastName;
+    user.username = updates.username !== undefined ? updates.username : user.username;
+    user.gender = updates.gender !== undefined ? updates.gender : user.gender; 
+    user.institution = updates.institution !== undefined ? updates.institution : user.institution;
+    user.country = updates.country !== undefined ? updates.country : user.country;
+    user.About = updates.About !== undefined ? updates.About : user.About;
+    user.phone = updates.phone !== undefined ? updates.phone : user.phone;
+    user.profileImage = updates.profileImage !== undefined ? updates.profileImage : user.profileImage;
 
     await user.save();
 
+    // fields needed in response
+    const responseData = {
+      firstName: user.firstName,
+      lastName: user.lastName,
+      username: user.username,
+      email: user.email,
+      profileImage: user.profileImage,
+      gender: user.gender,
+      role: user.role,
+      institution: user.institution,
+      country: user.country,
+      About: user.About,
+      phone: user.phone,
+      createdAt: user.createdAt,
+      updatedAt: user.updatedAt,
+      active: user.active,
+      verified: user.verified,
+    };
+
```

Comment:
Let's use sequelize to exclude sensitive data instead of using this manual approach